From 890e35ca84991ecd70dfdde396f53b9901d912bd Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 30 Apr 2020 12:49:24 -0500
Subject: [PATCH 24/25] arm64: dts: Enable PDM Mic

The baseboard has a PDM microphone.  This patch enables this port.

Record with the microphone using:

arecord -Dsysdefault:imxaudiomicfil -r 44100 -f S16_LE -c2 -d5 test.wave

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../freescale/beacon-imx8mm-baseboard.dtsi    | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
index d84c4d758f75..60348cd21660 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
@@ -148,6 +148,12 @@
 			"IN3R", "AMIC";
 	};
 
+	sound-micfil {
+		compatible = "fsl,imx-audio-micfil";
+		model = "imx-audio-micfil";
+		cpu-dai = <&micfil>;
+	};
+
 	sound-spdif {
 		compatible = "fsl,imx-audio-spdif";
 		model = "imx-spdif";
@@ -236,6 +242,13 @@
 			>;
 		};
 
+		pinctrl_pdm: pdmgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SAI5_RXC_PDM_CLK		0xd6
+				MX8MM_IOMUXC_SAI5_RXD0_PDM_DATA0	0xd6
+			>;
+		};
+
 		pinctrl_sai3: sai3grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI3_TXFS_SAI3_TX_SYNC     0xd6
@@ -376,6 +389,15 @@
 
 };
 
+&micfil {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pdm>;
+	assigned-clocks = <&clk IMX8MM_CLK_PDM>;
+	assigned-clock-parents = <&clk IMX8MM_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <196608000>;
+	status = "okay";
+};
+
 &mipi_csi_1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-- 
2.17.1

